### PR TITLE
BR: move check_cdc to public pkg

### LINKED ISSUE #41504: br restore is conflict with cdc changefeed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create cdc changefeed
2. start br restore
### 2. What did you expect to see? (Required)
restore should be failed
### 3. What did you see instead (Required)
restore success to run
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #41504 

Problem Summary:
BR restore should be failed when the cluster has cdc changefeed running
### What is changed and how it works?
Currently lightning has the step to check whether the cdc changefeed is running. Move the step to public pkg so that BR restore can directly use it.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/utils/cdc.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"go.uber.org/zap"
+)
+
+const (
+	CDCPrefix      = "/tidb/cdc/"
+	ChangefeedPath = "/changefeed/info/"
+	CDCPrefixV61   = "/tidb/cdc/changefeed/info/"
+)
+
+type CDCNameSet struct {
```
Comment (lance6716) [inline br/pkg/utils/cdc.go:22]: lint should complain that this structure has no comments

File: br/pkg/utils/cdc.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	nameSet map[string][]string
+}
+
+func (s *CDCNameSet) Empty() bool {
+	return len(s.nameSet) == 0
+}
+
+func (s *CDCNameSet) String() string {
```
Comment (lance6716) [inline br/pkg/utils/cdc.go:30]: I think this is not a String(). It should be named with MessageToUser or something. String() is a simple representation of inner map.

Comment (Leavrth) [reply br/pkg/utils/cdc.go]: done

File: br/pkg/utils/cdc.go (hunk 1 above)
Comment (Leavrth) [reply br/pkg/utils/cdc.go:24]: done

File: br/pkg/utils/cdc.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	return len(s.nameSet) == 0
+}
+
+// MessageToUser convert the map `nameSet` to a readable message to user.
+func (s *CDCNameSet) MessageToUser() string {
+	var changefeedMsgBuf strings.Builder
+	changefeedMsgBuf.WriteString("found CDC changefeed(s): ")
+	isFirst := true
+	for clusterID, captureIDs := range s.nameSet {
+		if !isFirst {
```
Comment (YuJuncen) [inline br/pkg/utils/cdc.go:39]: It seems we are going to yield string like `item1, item2, item3,`, why not just append a comma after each item like:
```
for _, item := range items {
  writer.WriteString(item);
  writer.WriteString(",");
}
```

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: br/pkg/utils/cdc.go (hunk 3 above)
Comment (Leavrth) [reply br/pkg/utils/cdc.go:39]: use `isFirst` to yield string like `item1, item2, item3`.

Comment (Leavrth) [reply br/pkg/utils/cdc.go]: done

**Pull request merged**